unitTests/ui/ResponsiveComponents/ResponsiveTabsTest.ts

```diff
@@ -33,5 +49,120 @@ export function ResponsiveTabsTest() {
+
+      describe('when the tab section is overflowing', () => {
+        beforeEach(() => {
+          simulateOverflowing();
+          spyOn(root, 'width').and.returnValue(new ResponsiveComponents().getMediumScreenWidth() - 1);
+        });
+
+        it('it creates a dropdown header', () => {
+          responsiveTabs.handleResizeEvent();
+          expect(root.find('.coveo-dropdown-header')).not.toBeNull();
+        });
+
+        it('places tabs in the correct order inside and outside the dropdown', () => {
+          responsiveTabs.handleResizeEvent();
+          const container = openTabContainer();
+
+          const lastTabInDropdown = container.find(`div[data-id="4"]`);
+          expect(lastTabInDropdown).not.toBeNull();
+
+          range(0, 4).forEach(otherTabsNotInsideDropdown => {
+            const otherTabWhichShouldNotBeInDropdown = container.find(`div[data-id="${otherTabsNotInsideDropdown}"]`);
+            expect(otherTabWhichShouldNotBeInDropdown).toBeNull();
+          });
+        });
+
+        it('should not place a selected tab inside the dropdown', () => {
+          last(tabs).addClass('coveo-selected');
+
+          responsiveTabs.handleResizeEvent();
+          const container = openTabContainer();
+
+          const selectedTabInDropdown = container.find(`div[data-id="4"]`);
+          expect(selectedTabInDropdown).toBeNull();
+        });
+      });
+
+      describe('when the tab section is overflowing, then resized to larger size', () => {
+        beforeEach(() => {
+          spyOn(root, 'width').and.returnValue(new ResponsiveComponents().getMediumScreenWidth() - 1);
+        });
+
+        it('replaces the tabs in the correct order in the original section', () => {
+          simulateOverflowing();
+          responsiveTabs.handleResizeEvent();
+          simulateNotOverflowing();
+          responsiveTabs.handleResizeEvent();
+
+          range(0, 5).forEach(tabThatShouldBeInStandardSection => {
+            expect(tabSection.find(`div[data-id="${tabThatShouldBeInStandardSection}"]`)).not.toBeNull();
+          });
+        });
+
+        it('should remove the dropdown header', () => {
+          simulateOverflowing();
+          responsiveTabs.handleResizeEvent();
+
+          expect(root.find('.coveo-dropdown-header')).not.toBeNull();
+
+          simulateNotOverflowing();
+          responsiveTabs.handleResizeEvent();
+
+          expect(root.find('.coveo-dropdown-header')).toBeNull();
+        });
+
+        it('replaces the tabs in the correct order in the original section when there was a selection', () => {
+          last(tabs).addClass('coveo-selected');
+
+          simulateOverflowing();
+          responsiveTabs.handleResizeEvent();
+          simulateNotOverflowing();
+          responsiveTabs.handleResizeEvent();
+
+          const allTabsReplacedInTheirOriginalSection = tabSection.findAll(`.${Component.computeCssClassName(Tab)}`);
+          range(0, 5).forEach(allTabsInExpectedOrder => {
```

Comment:
I think the variable here would be better named `tabId`. Then we could make it easier to read the `expect`. e.g

```
range(0, 5).forEach(tabId => {
    const tab = allTabsReplacedInTheirOriginalSection[tabId];
    expect(tab.getAttribute('data-id')).toEqual(tabId.toString());
})
```